packages/base/src/util/encodeXML.js

```diff
@@ -0,0 +1,35 @@
+const fnToHex = (iChar, iLength) => {
+	let sHex = iChar.toString(16);
+	if (iLength) {
+		sHex = sHex.padStart(iLength, "0");
+	}
+	return sHex;
+};
+
+const rHtml = /[\x00-\x2b\x2f\x3a-\x40\x5b-\x5e\x60\x7b-\xff\u2028\u2029]/g, // eslint-disable-line
+	rHtmlReplace = /[\x00-\x08\x0b\x0c\x0e-\x1f\x7f-\x9f]/, // eslint-disable-line
```

Comment:
**[nitpick (non-blocking)](https://conventionalcomments.org/#labels):** Per https://eslint.org/docs/user-guide/configuring/rules#disabling-rules, ideally, we would not disable all ESLint rules on these lines via `// eslint-disable-line`, and only disable the specific ones via `// eslint-disable-line XXX, XXX, XXX...` which the author of these lines purposely wanted to disable at the time of coding these lines, so that if and when in the future more ESLint rules might complain about these lines, we won't immediately disable these rules, without giving us an opportunity to see whether the rules are valid or not -- such that we might fix the issues rather than disable the new rules that identified the issues.